app/uk/gov/hmrc/customs/rosmfrontend/controllers/EnrolmentAlreadyExistsController.scala

```diff
@@ -38,11 +39,7 @@ class EnrolmentAlreadyExistsController @Inject()(
 
   override def messagesApi: MessagesApi = currentApp.injector.instanceOf[MessagesApi]
 
-  def enrolmentAlreadyExists(): Action[AnyContent] = Action.async { implicit request =>
-    authorised(AuthProviders(GovernmentGateway))
-      .retrieve(allEnrolments) { allEnrolments =>
-        val eoriNumber = allEnrolments.getEnrolment("HMRC-CUS-ORG").flatMap(_.getIdentifier("EORINumber").map(_.value))
-        Future.successful(Ok(registrationExistsView(eoriNumber)))
-      }
+  def enrolmentAlreadyExists(service: Service.Value): Action[AnyContent] = Action.async { implicit request =>
+    Future.successful(Ok(registrationExistsView(service)))
```

Comment:
So this controller now will be just displaying the page and all actions will be performed outside?

What do you think that instead of `Action.async` use and action to check if enrolment already exists? Because right now there is no restrictions to this page, so coming from cached url this page will be displayed

Edit: But as I see the previous implementation didn't check this, just retrieved the EORI

Reply:
Yes this is just a dumb "show the page" controller.  Doesn't have to by async anymore I guess.